rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -29,6 +33,10 @@
   private final TaskScheduler taskScheduler;
   private final ErrorHandler errorHandler;
 
+  // This list is used to retain error adapters in a non-weak reference associated with this use
+  // case handler
+  private final List<OnErrorCallback> errorCallbacks = new ArrayList<>();
```

Comment:
Giving we are in the context of Rosie, error callbacks won't be implemented by views but subclasses of `RosiePresenter` (or inner anonymous classes of it) that are already taking care of not holding strong references to GC roots. The only problem I see is that we will be retaining those callbacks for the lifetime of the presenter but I don't think that's such a problem honestly.